File: LeetCode/701-800/739. 每日温度.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> ans(n);
        stack<int> s;
        for (int i = 0; i < n; i++) {
            while(!s.empty() && temperatures[s.top()] < temperatures[i]) {
                int l = s.top();
                ans[l] = i - l;
                s.pop();
            }
            s.push(i);
        }
        return ans;
    }
};

File: LeetCode/701-800/739. 每日温度.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
       Stack<Integer> stack = new Stack<>();
        int n = temperatures.length;
        int[] res = new int[n];
        for (int i = 0; i < n; i++) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                Integer p = stack.pop();
                res[p] = i - p;
            }
            stack.push(i);
        }
        return res;
    }
}